src/libraries/Common/src/Interop/Browser/Interop.HttpHandlerService.cs

```diff
@@ -0,0 +1,535 @@
+            /// </summary>
+            private static bool StreamingSupported { get; }
+
+            public BrowserHttpHandlerService()
+            { }
+
+            static BrowserHttpHandlerService()
+            {
+                using (var streamingSupported = new Function("return typeof Response !== 'undefined' && 'body' in Response.prototype && typeof ReadableStream === 'function'"))
+                    StreamingSupported = (bool)streamingSupported.Call();
+                handlerInit();
+            }
+
+            private static void handlerInit()
+            {
+                window = (JSObject)Interop.Runtime.GetGlobalObject("window");
+                fetch = (JSObject)Interop.Runtime.GetGlobalObject("fetch");
+            }
+
+            public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                // There is a race condition on Safari as a result of using TaskCompletionSource that
+                // causes a stack exceeded error being thrown.  More information can be found here:
+                // https://devblogs.microsoft.com/premier-developer/the-danger-of-taskcompletionsourcet-class/
+                var tcs = new TaskCompletionSource<HttpResponseMessage>(TaskCreationOptions.RunContinuationsAsynchronously);
+                using (cancellationToken.Register(() => tcs.TrySetCanceled()))
+                {
+#pragma warning disable 4014
+                    doFetch(tcs, request, cancellationToken).ConfigureAwait(false);
+#pragma warning restore 4014
+
+                    return await tcs.Task.ConfigureAwait(true);
+                }
+            }
+
+            private async Task doFetch(TaskCompletionSource<HttpResponseMessage> tcs, HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                if (request == null)
+                {
+                    throw new ArgumentNullException(nameof(request));
+                }
+
+                try
+                {
+                    var requestObject = new JSObject();
+
+                    if (request.Properties.TryGetValue("WebAssemblyFetchOptions", out var fetchOoptionsValue) &&
+                        fetchOoptionsValue is IDictionary<string, object> fetchOptions)
+                    {
+                        foreach (var item in fetchOptions)
+                        {
+                            requestObject.SetObjectProperty(item.Key, item.Value);
+                        }
+                    }
+
+                    requestObject.SetObjectProperty("method", request.Method.Method);
+
+                    // We need to check for body content
+                    if (request.Content != null)
+                    {
+                        if (request.Content is StringContent)
+                        {
+                            requestObject.SetObjectProperty("body", await request.Content.ReadAsStringAsync().ConfigureAwait(true));
```

Comment:
ConfigureAwait(true)?  That's very uncommon.  Did you mean ConfigureAwait(false) for all of these?

Reply:
This was caught by the build:  `error CA2007: Consider calling ConfigureAwait on the awaited task` for all of these.  So basically to silence these errors.  Should I be using something else.  `ConfigureAwait(true)` is still the default correct?

Reply:
CA2007 is really suggesting to use `ConfigureAwait(false)`, though that's certainly not clear from the error message.  While I know it's annoying, pretty much every await in dotnet/runtime should use `.ConfigureAwait(false)`, and if it doesn't, there needs to be a good reason for it, generally called out in a comment.

Does any of this code have specific requirements as to what thread it runs on?

Reply:
Stephan thanks for the information and yes it really is not clear.  Wasm at this time is single threaded.  Will make that change.

Reply:
Addressed in recent commits

Reply:
Virtually all WASM ConfigureAwaits should be locked to the original thread, because most APIs only work on the main browser thread. If we continue on a worker thread any API calls will have to be remoted back to the main thread.

Reply:
> If we continue on a worker thread any API calls will have to be remoted back to the main thread.

What handles that remoting?

Is there a SynchronizationContext that's present that would queue continuations back, and is that different from the cited remoting?

Reply:
I don't know how we currently do it. Emscripten has a bodged together system for sending API calls from workers to the main thread, so if we're using their syscalls we might be using that. This will only matter once we enable threads, it's just important for us to figure it out now since the JS threading model is very odd.

Reply:
Ok.  If there's no SynchronizationContext (or TaskScheduler), then the value used for ConfigureAwait isn't going to matter, since it's only meaningful if there is one.  If there is one, I don't have a problem with us using ConfigureAwait(true) here, given the environment in which this type can be used, but we should have a comment explaining it.

Reply:
Right now I would say that there is an agreement that we should use `ConfigureAwait(continueOnCapturedContext: true)` in our library and allow the developer to set this how they want.  This forces the callback to be invoked on the original context or scheduler in our library code.

Will make this change as was modified to false in a previous commit.  Thanks for the comments Katelyn and Stephen.

Reply:
Addressed in recent commits